src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrariesAndRid.cs

```diff
@@ -91,26 +92,42 @@ public void It_builds_a_framework_dependent_RID_specific_runnable_output()
             var buildCommand = new BuildCommand(testAsset, "App");
 
             buildCommand
-                .Execute($"/p:RuntimeIdentifier={runtimeIdentifier}", $"/p:TestRuntimeIdentifier={runtimeIdentifier}", "/p:SelfContained=false")
+                .Execute($"/p:RuntimeIdentifier={runtimeIdentifier}", $"/p:TestRuntimeIdentifier={runtimeIdentifier}", "/p:SelfContained=false", "/p:ProduceReferenceAssembly=false")
                 .Should().Pass();
 
-            var outputDirectory = buildCommand.GetOutputDirectory("netcoreapp2.1", runtimeIdentifier: runtimeIdentifier);
+            var outputDirectory = buildCommand.GetOutputDirectory(ToolsetInfo.CurrentTargetFramework, runtimeIdentifier: runtimeIdentifier);
 
             outputDirectory.Should().NotHaveSubDirectories();
-            outputDirectory.Should().OnlyHaveFiles(new[] {
+
+            var sqlFile = "";
+            if (OperatingSystem.IsWindows())
```

Comment:
Generally we use `RuntimeInformation.IsOSPlatform` to do OS checks, which I thought was the "right" way to do it.  It looks like new APIs were added to `OperatingSystem` in .NET 5, though, so maybe it's now fine to use that.

Reply:
They actually recommended we use the `OperatingSystem` ones as they are faster: https://github.com/dotnet/sdk/issues/12825